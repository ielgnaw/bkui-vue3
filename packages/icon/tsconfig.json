{
  "ts-node": {
    "transpileOnly": true
  },
  "compilerOptions": {
    "target": "ES2019",
    "module": "commonjs",
    "strict": true,
    "declaration": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "jsx": "preserve",
    "jsxFactory": "h"
  },
  "include": [
    "./src/**/*",
    "./scripts/*/*"
  ],
  "exclude": [
    "*.js"
  ]
}