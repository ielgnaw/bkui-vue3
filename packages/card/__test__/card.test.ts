/*
 * Tencent is pleased to support the open source community by making
 * 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community Edition) available.
 *
 * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
 *
 * 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community Edition) is licensed under the MIT License.
 *
 * License for 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community Edition):
 *
 * ---------------------------------------------------
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
 * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
 * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
 * the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
 * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
 * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */

import { Help, HelpDocumentFill, HelpFill } from '@bkui-vue/icon';
import { mount } from '@vue/test-utils';

import Card from '../src';
const mountTpl = (template: string) =>
  mount(
    {
      components: {
        bkCard: Card,
        Help,
        HelpDocumentFill,
        HelpFill,
      },
      template,
    },
    {
      global: {
        provide: {
          card: {},
        },
      },
    },
  );
describe('Card.tsx', () => {
  it('showHeader', async () => {
    const wrapper = await mount(Card, {
      props: { showHeader: false },
    });
    expect(wrapper.findAll('.bk-card-head').length).toEqual(0);
  });
  it('showFooter && hideHeader ', async () => {
    const wrapper = await mount(Card, {
      props: { showHeader: false, showFooter: true },
    });
    expect(wrapper.findAll('.bk-card-head').length).toEqual(0);
    expect(wrapper.findAll('.bk-card-footer').length).toEqual(1);
  });
  it('showFooter', async () => {
    const wrapper = await mount(Card, {
      props: { showFooter: true },
    });
    expect(wrapper.findAll('.bk-card-footer').length).toEqual(1);
  });
  it('border', async () => {
    const wrapper = await mount(Card, {
      props: { border: false },
    });
    expect(wrapper.findAll('.bk-card-border-none').length).toEqual(1);
  });
  it('headBorder', async () => {
    const wrapper = await mount(Card, {
      props: { headBorder: false },
    });
    expect(wrapper.findAll('.bk-card-border-none').length).toEqual(1);
  });
  it('title', async () => {
    const wrapper = await mount(Card, {
      props: { title: 'Card Title' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('collapseStatus', async () => {
    const wrapper = await mount(Card, {
      props: { collapseStatus: false },
    });
    expect(wrapper.findAll('.bk-card-body').length).toEqual(0);
  });
  it('default slot', async () => {
    const wrapper = mountTpl(`
    <bkCard>
      <div>
          <p>卡片内容 1</p>
          <p>卡片内容 2</p>
          <p>卡片内容 3</p>
      </div>
    </bkCard>`);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('footer slot', async () => {
    const wrapper = mountTpl(`
    <bkCard>
      <div>
          <p>卡片内容 1</p>
          <p>卡片内容 2</p>
          <p>卡片内容 3</p>
      </div>
      <div slot="footer">
        <div style="background: #fafbfd;">
          <span class="card-foot-icon"><HelpDocumentFill /></span>
          <span class="card-foot-icon"><Help /></span>
          <span class="card-foot-icon"><HelpFill /></span>
        </div>
      </div>
    </bkCard>`);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('Header slot', async () => {
    const wrapper = mountTpl(`
    <bkCard :showHeader=${true} :showFooter=${false}>
      <div slot="header">
        <Help />
        <label>Card Test</label>
      </div>
      <div>
          <p>卡片内容 1</p>
          <p>卡片内容 2</p>
          <p>卡片内容 3</p>
      </div>
    </bkCard>`);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
