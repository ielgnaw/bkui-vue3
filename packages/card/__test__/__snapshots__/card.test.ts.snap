// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.tsx Header slot 1`] = `
"<div class=\\"bk-card\\">
  <div class=\\"bk-card-head\\">
    <!---->
    <div class=\\"title\\">
      <!---->
    </div>
    <!---->
  </div>
  <div>
    <div class=\\"bk-card-body\\">
      <div slot=\\"header\\"><span><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"bk-icon\\" style=\\"vertical-align: middle; fill: undefined; overflow: hidden;\\" overflow=\\"hidden\\" viewBox=\\"0 0 1024 1024\\"><path d=\\"M512 128c212.8 0 384 171.2 384 384S724.8 896 512 896 128 724.8 128 512 299.2 128 512 128M512 64C264 64 64 264 64 512s200 448 448 448 448-200 448-448S760 64 512 64z\\" style=\\"\\"></path><path d=\\"M548.8 673.6A48 48 0 0 1 500.8 721.6 48 48 0 0 1 452.8 673.6 48 48 0 0 1 548.8 673.6z\\" style=\\"\\"></path><path d=\\"M513.6 302.4c80 0 132.8 44.8 132.8 110.4 0 41.6-20.8 72-60.8 96-40 24-48 35.2-48 62.4v14.4H464v-16c-3.2-44.8 11.2-70.4 51.2-92.8 36.8-22.4 48-35.2 48-60.8s-20.8-44.8-52.8-44.8c-28.8-1.6-52.8 19.2-56 48 0 1.6 0 1.6 0 3.2h-76.8C377.6 352 427.2 302.4 513.6 302.4z\\" style=\\"\\"></path></svg></span><label>Card Test</label></div>
      <div>
        <p>卡片内容 1</p>
        <p>卡片内容 2</p>
        <p>卡片内容 3</p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Card.tsx default slot 1`] = `
"<div class=\\"bk-card\\">
  <div class=\\"bk-card-head\\">
    <!---->
    <div class=\\"title\\">
      <!---->
    </div>
    <!---->
  </div>
  <div>
    <div class=\\"bk-card-body\\">
      <div>
        <p>卡片内容 1</p>
        <p>卡片内容 2</p>
        <p>卡片内容 3</p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Card.tsx footer slot 1`] = `
"<div class=\\"bk-card\\">
  <div class=\\"bk-card-head\\">
    <!---->
    <div class=\\"title\\">
      <!---->
    </div>
    <!---->
  </div>
  <div>
    <div class=\\"bk-card-body\\">
      <div>
        <p>卡片内容 1</p>
        <p>卡片内容 2</p>
        <p>卡片内容 3</p>
      </div>
      <div slot=\\"footer\\">
        <div style=\\"background: rgb(250, 251, 253);\\"><span class=\\"card-foot-icon\\"><span><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"bk-icon\\" style=\\"vertical-align: middle; fill: undefined; overflow: hidden;\\" overflow=\\"hidden\\" viewBox=\\"0 0 1024 1024\\"><path d=\\"M512 64C264 64 64 264 64 512c0 57.6 11.2 113.6 32 166.4V896c0 17.6 14.4 32 32 32h217.6C576 1019.2 836.8 908.8 928 678.4S908.8 187.2 678.4 96C625.6 75.2 569.6 64 512 64zM500.8 721.6c-27.2 0-48-20.8-48-48s20.8-48 48-48c27.2 0 48 20.8 48 48S528 721.6 500.8 721.6zM587.2 507.2c-36.8 20.8-48 35.2-48 62.4v14.4H464v-16c-3.2-44.8 11.2-70.4 51.2-92.8 36.8-22.4 48-35.2 48-60.8s-20.8-44.8-52.8-44.8c-28.8-1.6-52.8 19.2-56 48 0 1.6 0 1.6 0 3.2h-76.8c1.6-70.4 49.6-118.4 136-118.4 80 0 132.8 44.8 132.8 110.4C648 454.4 627.2 484.8 587.2 507.2z\\" style=\\"\\"></path></svg></span></span><span class=\\"card-foot-icon\\"><span><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"bk-icon\\" style=\\"vertical-align: middle; fill: undefined; overflow: hidden;\\" overflow=\\"hidden\\" viewBox=\\"0 0 1024 1024\\"><path d=\\"M512 128c212.8 0 384 171.2 384 384S724.8 896 512 896 128 724.8 128 512 299.2 128 512 128M512 64C264 64 64 264 64 512s200 448 448 448 448-200 448-448S760 64 512 64z\\" style=\\"\\"></path><path d=\\"M548.8 673.6A48 48 0 0 1 500.8 721.6 48 48 0 0 1 452.8 673.6 48 48 0 0 1 548.8 673.6z\\" style=\\"\\"></path><path d=\\"M513.6 302.4c80 0 132.8 44.8 132.8 110.4 0 41.6-20.8 72-60.8 96-40 24-48 35.2-48 62.4v14.4H464v-16c-3.2-44.8 11.2-70.4 51.2-92.8 36.8-22.4 48-35.2 48-60.8s-20.8-44.8-52.8-44.8c-28.8-1.6-52.8 19.2-56 48 0 1.6 0 1.6 0 3.2h-76.8C377.6 352 427.2 302.4 513.6 302.4z\\" style=\\"\\"></path></svg></span></span><span class=\\"card-foot-icon\\"><span><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"bk-icon\\" style=\\"vertical-align: middle; fill: undefined; overflow: hidden;\\" overflow=\\"hidden\\" viewBox=\\"0 0 1024 1024\\"><path d=\\"M512 64C264 64 64 264 64 512s200 448 448 448 448-200 448-448S760 64 512 64zM500.8 721.6c-27.2 0-48-20.8-48-48s20.8-48 48-48c27.2 0 48 20.8 48 48S528 721.6 500.8 721.6zM587.2 507.2c-36.8 20.8-48 35.2-48 62.4v14.4H464v-16c-3.2-44.8 11.2-70.4 51.2-92.8 36.8-22.4 48-35.2 48-60.8s-20.8-44.8-52.8-44.8c-28.8-1.6-52.8 19.2-56 48 0 1.6 0 1.6 0 3.2h-76.8c1.6-70.4 49.6-118.4 136-118.4 80 0 132.8 44.8 132.8 110.4C648 454.4 627.2 484.8 587.2 507.2z\\" style=\\"\\"></path></svg></span></span></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Card.tsx title 1`] = `
"<div class=\\"bk-card\\">
  <div class=\\"bk-card-head\\">
    <!---->
    <div class=\\"title\\" title=\\"Card Title\\">Card Title</div>
    <!---->
  </div>
  <div>
    <div class=\\"bk-card-body\\">Content</div>
  </div>
</div>"
`;
